### PR TITLE
[WIP] Bootstrapping Google O-auth Login

### PR DESCRIPTION
**Pull request by himanshuc3**
## Adding Google O-auth.

It's not completely working. We can discuss problems/shortcomings related to this PR here and eventually I could work on Login to be more precise.

 

- [ ] 1 error is occuring **during callback from auth**:

```
Error
   at Strategy.OAuth2Strategy.parseErrorResponse (/Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/passport-oauth2/lib/strategy.js:329:12)
   at Strategy.OAuth2Strategy._createOAuthError (/Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/passport-oauth2/lib/strategy.js:376:16)
   at /Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/passport-oauth2/lib/strategy.js:166:45
   at /Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/oauth/lib/oauth2.js:191:18
   at passBackControl (/Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/oauth/lib/oauth2.js:132:9)
   at IncomingMessage.<anonymous> (/Users/himanshu/Desktop/p5/p5.js-web-editor/node_modules/oauth/lib/oauth2.js:157:7)
   at emitNone (events.js:91:20)
   at IncomingMessage.emit (events.js:188:7)
   at endReadableNT (_stream_readable.js:975:12)
   at _combinedTickCallback (internal/process/next_tick.js:80:11)
   at process._tickDomainCallback (internal/process/next_tick.js:128:9) 
```

### CODE REVIEW

**Comment by catarak** [comment]
this is WIP (work in progress)?

**Comment by himanshuc3** [comment]
Yes @catarak 
I've mentioned an error message in the description. I've searched google but nothing seems to work. Can you help me with this?

File: server/config/passport.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ const lodash = require('lodash');
 const passport = require('passport');
 const GitHubStrategy = require('passport-github').Strategy;
 const LocalStrategy = require('passport-local').Strategy;
+const GoogleStrategy = require('passport-google-oauth2').Strategy;
+const obj = require('./id');
```
Comment (catarak) [inline server/config/passport.js]: you should add the google key/secret key to the `.env` file!

**Comment by catarak** [comment]
it looks like the google oauth is returning an error (given `Strategy.OAuth2Strategy.parseErrorResponse`). Maybe it's not being caught/handled properly so it's unclear what the error is?

**Comment by himanshuc3** [comment]
ok, I'll look into it.

**Pull request closed without merging**